src/pages/taskHistory/index.tsx

```diff
@@ -38,33 +40,58 @@ const TaskHistoryContents: React.VFC = () => {
     projectIdentifier: string;
     taskName: string;
   }>();
+  const { ingestNewCommits } = useHistoryTable();
   usePageTitle(`Task History | ${projectIdentifier} | ${taskName}`);
-  const [nextPageOrderNumber, setNextPageOrderNumber] = useState(null);
   useTestFilters();
   useJumpToCommit();
 
   const { badges, handleOnRemove, handleClearAll } = useFilterBadgeQueryParams(
     constants.queryParamsToDisplay
   );
-  const { data, error } = useQuery<
+  const dispatchToast = useToastContext();
+
+  const { data, loading, refetch } = useQuery<
     MainlineCommitsForHistoryQuery,
     MainlineCommitsForHistoryQueryVariables
   >(GET_MAINLINE_COMMITS_FOR_HISTORY, {
     variables: {
       mainlineCommitsOptions: {
         projectIdentifier,
         limit: 10,
```

Comment:
Would increasing the limit here have a big hit on performance? Having to fetch so frequently feels noticeable when scrolling imo.

Reply:
https://jira.mongodb.org/browse/EVG-20217